core/server/master/src/main/java/alluxio/master/file/meta/MountTable.java

```diff
@@ -234,6 +228,17 @@ public boolean delete(Supplier<JournalContext> journalContext, AlluxioURI uri) {
 
     try (LockResource r = new LockResource(mWriteLock)) {
       if (mState.getMountTable().containsKey(path)) {
+        // check if the path contains another nested mount point
+        for (String mountPath : mState.getMountTable().keySet()) {
+          try {
+            if (PathUtils.hasPrefix(mountPath, path) && (!path.equals(mountPath))) {
+              LOG.warn("The path to unmount contains another nested mountpoint");
+              return false;
+            }
+          } catch (InvalidPathException e) {
+            LOG.warn("Invalid path encountered when checking for nested mount point");
```

Comment:
Would be helpful to add the invalid path string and the exception to the log entry.

Reply:
done